services/QuillLMS/client/app/bundles/Teacher/components/assignment_flow/manage_units/manage_units.jsx

```diff
@@ -38,8 +39,14 @@ export default class ManageUnits extends React.Component {
   }
 
   getClassrooms = () => {
+    const { selectedClassroomId, } = this.state
     request.get(`${process.env.DEFAULT_URL}/teachers/classrooms/classrooms_i_teach`, (error, httpStatus, body) => {
       const classrooms = JSON.parse(body).classrooms;
+      const localStorageSelectedClassroomId = window.localStorage.getItem(PROGRESS_REPORTS_SELECTED_CLASSROOM_ID)
```

Comment:
You could, but since it is only ever getting used once either way I don't think there's a performance benefit, and I think since the point of converting it is for comparison's sake to me it seems more natural to do it when you are actually doing the comparison.